### ISSUE TITLE
Issue with TypeVar and classmethod

### ISSUE THREAD

**Issue by bloussou** [microsoft/pyright#1500]
**Describe the bug**
The bug is described in this discussion https://github.com/microsoft/pyright/discussions/1499. In the code to reproduce, `cls._create({})` raise this error : `Expression of type "Configuration" cannot be assigned to return type "_Configuration@_from_dict" Type "Configuration" cannot be assigned to type "_Configuration@_from_dict" Pyright(reportGeneralTypeIssues)`

**To Reproduce**
```
from typing import Any, Mapping, Type, TypeVar

_Configuration = TypeVar("_Configuration", bound="Configuration")


class Configuration:
    @classmethod
    def _create(cls: Type[_Configuration], data: Mapping[str, Any]) -> _Configuration:
        ...

    @classmethod
    def _from_dict(
        cls: Type[_Configuration], data: Mapping[str, Any]
    ) -> _Configuration:
        return cls._create({})
```

**Expected behavior**
`Configuration._from_dict` must have `Configuration` as returned type.

**Cli or VsCode**
This happen both in the cli and VsCode after version `1.102` (included)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.113, which I just published. It will also be included in the next release of Pylance.
